TAO/TAO_IDL/util/utl_err.cpp

```diff
@@ -185,9 +185,11 @@ error_string (UTL_Error::ErrorCode c)
     case UTL_Error::EIDL_KEYWORD_WARNING:
       return "Warning - spelling differs from IDL keyword only in case: ";
     case UTL_Error::EIDL_ANONYMOUS_ERROR:
-      return "Error: anonymous types are deprecated by OMG spec";
+      return "Error: anonymous types are deprecated by OMG spec in IDL3, "
+        "but are valid again in IDL4";
     case UTL_Error::EIDL_ANONYMOUS_WARNING:
-      return "Warning - anonymous types are deprecated by OMG spec";
+      return "Warning - anonymous types are deprecated by OMG spec in IDL3, "
+        "but are valid in IDL4";
```

Comment:
Maybe add "but not supported in all IDL language mappings"

Reply:
And/Or... "in those Profiles of IDL4 which include the Anonymous Types Building Block"

Or, since utl_err.cpp is part of TAO_IDL_FE, this could be changed to customize the error/warning messages based on which backend is used.  The backends are more directly related to language mappings (although this is not an entirely clean separation, it's enough to be useful).

Reply:
I don't really want to do anything else here. I think it's up to the backend to complain about typedefs if it doesn't support them, which it can do easily when it encounters a `AST_Typedef`.

Reply:
I mixed that up, I meant that I'm assuming the fact that it's an anonymous type can be detected by the backend. If that's not possible or easy, then I'm fine with making changes for that.